### PR TITLE
Subtype enforce for yml files

### PR DESCRIPTION
**Pull request by bakatzir**
## Status
Ready

## Related Issues
fixes: https://github.com/demisto/etc/issues/18582

## Description
add subtype check for yml files of type python

### CODE REVIEW

File: Tests/scripts/hook_validations/integration.py [hunk 1]
```diff
@@ -52,9 +52,26 @@ def is_backward_compatible(self):
         self.is_changed_command_name_or_arg()
         self.is_there_duplicate_args()
         self.is_there_duplicate_params()
+        self.is_valid_subtype()
 
         return self._is_valid
 
+    def is_valid_subtype(self):
+        """Validate that the subtype self.file_path is -1."""
```
Comment (ronykoz) [inline Tests/scripts/hook_validations/integration.py]: Change the -1 to python2/3

File: Tests/scripts/hook_validations/integration.py [hunk 2]
```diff
@@ -52,9 +52,26 @@ def is_backward_compatible(self):
...
+            if not subtype or subtype not in ['python3', 'python2']:
+                print_error("The subtype for our yml files should be either python2 or python3, "
+                            "please update the file {}.".format(self.current_integration.get('name')))
+                self._is_valid = False
+            old_subtype = self.old_integration.get('script', {}).get('subtype')
+            if old_subtype and old_subtype != subtype:
+                print_error("Possible backwards compatibility break, You've changed the subtype"
+                            " of the file {}".format(self.file_path))
+                self._is_valid = False
+        return not self._is_valid
```
Comment (ronykoz) [inline Tests/scripts/hook_validations/integration.py]: Why do you have the not? 
And no need for this line

File: Tests/scripts/hook_validations/tests/integration_test.py [hunk 3]
```diff
@@ -568,3 +568,16 @@ def test_duplicated_argss():
...
+    validator = IntegrationValidator("temp_file", check_git=False)
+    validator.current_integration = {
+        "script": {
+            "type": "python",
+            "subtype": "python3"
+        }
+    }
+
+    assert validator.is_valid_subtype(), \
+        "The integration validator did not find invalid subtype although there is"
```
Comment (ronykoz) [inline Tests/scripts/hook_validations/tests/integration_test.py]: Should be “found invalid subtype while it is valid”

File: Tests/scripts/hook_validations/tests/script_test.py [hunk 4]
```diff
@@ -440,3 +440,14 @@ def test_configuration_extraction():
...
+
+def test_valid_subtype():
+    validator = ScriptValidator("temp_file", check_git=False)
+    validator.current_script = {
+        "type": "python",
+        "subtype": "python3"
+    }
+
+    assert validator.is_valid_subtype(), \
+        "The script validator did not find invalid subtype although there is"
```
Comment (ronykoz) [inline Tests/scripts/hook_validations/tests/script_test.py]: Same here

**Comment by bakatzir** [comment]
@ronykoz will there always be an old_integration/script?
if no, how to handle?

**Comment by ronykoz** [comment]
> @ronykoz will there always be an old_integration/script?
> if no, how to handle?

@bakatzir add a condition for self.old_integration/script - and if they exist only then make the subtype comparison

**Pull request merged**